packages/jest-circus/src/utils.js

```diff
@@ -0,0 +1,225 @@
+/**
```

Comment:
what exactly do you not like about util methods?
these are actual pure utility functions that transform/massage the data. I took them out of the rest of the code because they:
1. are mostly stable and don't need to be changed
2. have too much noise (if/else, branching, loops, etc.)
3. should almost never be looked at.

I really dislike colocating them with the files because after a few refactors they end up being coupled to the rest of the code and we end up with spaghetti code that's very hard to read :(